README.md

```diff
@@ -672,119 +550,144 @@ r.Resource("/timeout",
+By default, a `ReadHeaderTimeout` of _10 seconds_ and an `IdleTimeout` of _15 seconds_ are set at the server level and apply to every incoming request.
 
-When triggered, the server sends a 408 Request Timeout as JSON with a message containing the time waited.
+Each operation's individual read timeout defaults to _15 seconds_ and can be changed as needed. This enables large request and response bodies to be sent without fear of timing out, as well as the use of WebSockets, in an opt-in fashion with sane defaults.
+
+When using the built-in model processing and the timeout is triggered, the server sends an error as JSON with a message containing the time waited.
 
 ```go
 type Input struct {
-	ID string
+	ID string `json:"id"`
 }
 
-r := huma.NewRouter("My API", "1.0.0")
+app := cli.NewRouter("My API", "1.0.0")
+foo := app.Resource("/foo")
 
 // Limit to 5 seconds
-r.Resource("/foo", huma.BodyReadTimeout(5 * time.Second)).Post(
-	"Create item", func(input *Input) string {
-		return "Hello, " + input.ID
-	})
+create := foo.Post("create-item", "Create a new item",
+	responses.NoContent(),
+)
+create.BodyReadTimeout(5 * time.Second)
+create.Run(func (ctx huma.Context, input Input) {
+	// Do something here.
+})
 ```
 
 You can also access the underlying TCP connection and set deadlines manually:
 
 ```go
-r.Resource("/foo", huma.ConnDependency()).Get(func (conn net.Conn) string {
+create.Run(func (ctx huma.Context, input struct {
+	Body io.Reader
+}) {
+	// Get the connection.
+	conn := huma.GetConn(ctx)
+
 	// Set a new deadline on connection reads.
 	conn.SetReadDeadline(time.Now().Add(600 * time.Second))
 
 	// Read all the data from the request.
-	data, err := ioutil.ReadAll(c.Request.Body)
+	data, err := ioutil.ReadAll(input.Body)
 	if err != nil {
+		// If a timeout occurred, this will be a net.Error with `err.Timeout()`
+		// returning true.
 		panic(err)
 	}
 
-	// Do something with the data...
-	return fmt.Sprintf("Read %d bytes", len(data))
+	// Do something with data here...
 })
 ```
 
-> :whale: Set to `-1` in order to disable the timeout.
+> :whale: Use `NoBodyReadTimeout()` to disable the default.
 
-### Request Body Size Limits
+#### Request Body Size Limits
 
 By default each operation has a 1 MiB reqeuest body size limit.
 
-When triggered, the server sends a 413 Request Entity Too Large as JSON with a message containing the maximum body size for this operation.
+When using the built-in model processing and the timeout is triggered, the server sends an error as JSON with a message containing the maximum body size for this operation.
 
 ```go
-r := huma.NewRouter("My API", "1.0.0")
+app := cli.NewRouter("My API", "1.0.0")
 
+create := app.Resource("/foo").Post("create-item", "Create a new item",
+	responses.NoContent(),
+)
 // Limit set to 10 MiB
-r.Resource("/foo", MaxBodyBytes(10 * 1024 * 1024)).Get(...)
+create.MaxBodyBytes(10 * 1024 * 1024)
```

Comment:
You get this (serialized to JSON in the response), since we stop reading and may not know the full input size. Length might come from the request's `Content-Length` header or it might be zero if unknown:

```go
ctx.AddError(&ErrorDetail{
  Message:  fmt.Sprintf("Request body too large, limit = %d bytes", ctx.op.maxBodyBytes),
  Location: "body",
  Value:    length,
})
```

From https://github.com/danielgtaylor/huma/pull/11/files#diff-6756fdfda338c13b910a6416f2ac9f54R194